docs/rhoas/iac_tools_rhoas/README.adoc

```diff
@@ -0,0 +1,703 @@
+- SSO_BASE_HOST - The base host for the SSO. This is the base URL for the SSO. For example, https://sso.redhat.com/auth/realms/redhat-external.
+
+.Procedure
+
+. Once the collection is installed, documentation for each module can be found by entering `ansible-doc rhoas.rhoas.<module_name>` and replacing <module_name> with the name of the module. The following example outputs the documentation for the `create_kafka` module. It shows the required (denoted by `=`) and optional parameters (denoted by `-`) for the module, along with examples of how to use the module, and the output of the module.
++
+[source,shell]
+----
+$ ansible-doc rhoas.rhoas.create_kafka
+----
+. You can run a single module to complete one task like in the following example which creates a Kafka instance.
++
+.Example create_kafka module
+[source,shell]
+----
+$ ansible localhost -m rhoas.rhoas.create_kafka -a 'name=unique-kafka-name billing_model=standard cloud_provider=aws plan="developer.x1" region="us-east-1" openshift_offline_token=<OFFLINE_TOKEN>'
+----
+. Alternatively, you can run a playbook. Enter the following command to run the https://github.com/redhat-developer/app-services-ansible/blob/main/rhoas_test.yml[rhoas test playbook].
++
+[source,shell]
+----
+$ ansible-playbook rhoas_test.yml
+----
+[NOTE]
+The openshift_offline_token is used for authentication to enable communication with the Kafka Management API. If the token is not passed in as an argument per task, the module attempts to read it from the `_OFFLINE_TOKEN_` environment variable.
+
++
+NOTE: The example playbook used in this section includes comments that indicate how to directly specify values rather than fetching them dynamically. For example, to specify a Kafka instance ID, a comment in the playbook states that you can include the following line:
++
+[source]
+----
+kafka_id: <kafka_id>
+----
+
++
+
+This playbook, if run as is, will complete the following tasks using these corresponding modules:
++
+.. Provision a {product-kafka} instance using the `create_kafka` module. The only required fields for this module are the `name`, `billing_model` and `cloud provider` fields. You can specify configuration options such as `billing_cloud_account_id` and the `openshift_offline_token`. All other information for the instance is provided by the Kafka APIs.
++
+[NOTE]
+When a module is run, Ansible saves the output of that command in a variable in the `register` field. In the following example, Ansible saves the created Kafka instance as `_kafka_req_resp_`.
++
+
+.Example `create_kafka` module
+[source,yaml]
+----
+- name: Create kafka
+    rhoas.rhoas.create_kafka:
+      name: "kafka-name"
+      instance_type: "x1"
+      billing_model: "standard"
+      cloud_provider: "aws"
+      region: "us-east-1"
+      plan: "developer.x1"
+      billing_cloud_account_id: "123456789"
+      openshift_offline_token: "OPENSHIFT_OFFLINE_TOKEN"
+    register:
+      kafka_req_resp
+----
```

Comment:
I was mulling over how you might best structure the next set of steps. I think it might be a good idea to add a step that instructs the user to open the example playbook and see the various modules that are in it. Then, each subsequent job can be about configuring each module, or at least understanding the default configuration. I've given this a stab below:

```
. In your web browser, navigate to https://github.com/redhat-developer/app-services-ansible/blob/main/rhoas_test.yml to open the OpenShift Application Services example playbook.

. Inspect the contents of the example playbook. In particular, observe that the playbook has modules for the following tasks:

* Creating and deleting a Kafka instance
* Creating and deleting a service account
* Creating Access Control List (ACL) permission bindings
* Creating, updating, and deleting a topic

. Configure the `create_kafka_` module to provision a new Kafka instance. The only required fields for this module are the `name`, `billing_model` and `cloud provider` fields. You can specify configuration options such as `billing_cloud_account_id` and the `openshift_offline_token`. All other information for the instance is provided by the Kafka APIs. The default contents of this module in the OpenShift Application Services example playbook are shown below:
+
.Example `create_kafka` module
[source,yaml]
----
- name: Create kafka
    rhoas.rhoas.create_kafka:
      name: "kafka-name"
      instance_type: "x1"
      billing_model: "standard"
      cloud_provider: "aws"
      region: "us-east-1"
      plan: "developer.x1"
      billing_cloud_account_id: "123456789"
      openshift_offline_token: "OPENSHIFT_OFFLINE_TOKEN"
    register:
      kafka_req_resp
----
+
When you run the `create_kafka_` module, Ansible saves the output of that command in a variable in the `register` field. In the preceding example, Ansible saves the created Kafka instance as `_kafka_req_resp_`